crates/ruff/tests/format.rs

```diff
@@ -2086,3 +2086,31 @@ fn range_formatting_notebook() {
     error: Failed to format main.ipynb: Range formatting isn't supported for notebooks.
     ");
 }
+
+/// Test that the formatter respects `per-file-target-version`. Context managers can't be
+/// parenthesized like this before Python 3.10.
+///
+/// Adapted from <https://github.com/python/cpython/issues/56991#issuecomment-1093555135>
+#[test]
+fn per_file_target_version_formatter() {
+    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME))
+        .args(["format", "--isolated", "--stdin-filename", "test.py", "--target-version=py38"])
+        .args(["--config", r#"per-file-target-version = {"test.py" = "py311"}"#])
+        .arg("-")
+        .pass_stdin(r#"
+with open("a_really_long_foo") as foo, open("a_really_long_bar") as bar, open("a_really_long_baz") as baz:
+    pass
+"#), @r#"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    with (
+        open("a_really_long_foo") as foo,
+        open("a_really_long_bar") as bar,
+        open("a_really_long_baz") as baz,
+    ):
+        pass
+
+    ----- stderr -----
+    "#);
```

Comment:
Same as for the linter. I think it would be good to have two assertions in this test:

1. One that asserts how the code gets formatted without the `--per-file-target-version` configuration set
2. One that asserts how the code gets formatted **with** the `--per-file-target-version` set. 


This ensures that the formatting change is, in fact, only due to the `--per-file-target-version` configuration and not because the formatter detected any other Python 3.9 syntax in the file and, therefore, decided that it's fine to parenthesize with items.